### PR TITLE
[air] Fix Tune checkpoint tracking for minimizing metrics

### PR DESCRIPTION
**Pull request by krfricke**
Signed-off-by: Kai Fricke <kai@anyscale.com>

## Why are these changes needed?

We're currently passing the checkpoint score attribute to the tune checkpoint manager without adding the legacy prefix `min-` if we want to minimize the metric. This leads to faulty tracking of checkpoints. In this PR, we update Tune's checkpoint manager to accept the full checkpoint config directly, avoiding problems in converting between the legacy format and the new API.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by dmatrix** [comment]
Brilliant! I ran into this bug today. I can provide a reproducible case if you want.

File: python/ray/air/tests/test_checkpoint_config.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        session.report(
+            dict(loss=7), checkpoint=Checkpoint.from_dict({"idx": 2})
+        )  # worst, deleted
+        session.report(dict(loss=5), checkpoint=Checkpoint.from_dict({"idx": 3}))
+
+    checkpoint_config = CheckpointConfig(
+        num_to_keep=2, checkpoint_score_attribute="loss", checkpoint_score_order="min"
+    )
+
+    trainer = DataParallelTrainer(
```
Comment (amogkam) [inline python/ray/air/tests/test_checkpoint_config.py:36]: can we move this test to `test_data_parallel_trainer`?

**Review by xwjiang2010 — APPROVED** [review]
